### DISCUSSION TITLE
Proposal: Integrating Slides Feature Directly into Observable Framework

### DISCUSSION THREAD

**Discussion by eli-s-goldberg** [observablehq/framework#1613] [Q&A] [answered]
Hi Observable Team,

I've noticed that Observable includes a couple of slide implementations, such as [Mike Freeman's Slides](https://observablehq.com/@mkfreeman/slides) and [Visnu Pitiyanuvath's Slides](https://observablehq.com/@visnup/slides). 

These are great, but integrating a slide feature directly into the Observable framework could benefit users significantly by standardizing and streamlining the process.

Would anyone else find a native slide feature in Observable useful? It seems like integrating this directly into the framework could save time and effort, especially for those of us who frequently use slides for presentations. I'm curious to hear your thoughts—do you think this would enhance your workflow, or do you prefer the flexibility of the current, more modular approach?

**Comment by mcmcclur** [comment, marked as answer]
I agree this wold be nice. Since I'm not sure when or if that might happen, though, reading this post motivated me to try to integrate RevealJS into a Framework project, which I've just set up here:

https://mcmcclur.observablehq.cloud/mark-on-framework/posts/RevealJS/

That that might be worth mentioning.

**Reply by eli-s-goldberg** [reply]
Love it! 

Will try to follow your example and report back.

**Reply by Fil** [reply]
@mcmcclur this is beautiful — is there a way to activate fullscreen?

**Reply by mcmcclur** [reply]
Yes! Just press the “f” key.

**Reply by Fil** [reply]
Ha now I see it's written on the first slide! Sorry for the noise :)

**Reply by Fil** [reply]
I love the way the bar chart fades in, then updates with scrolling. I see the `reveal.on('slidechanged', function(evt) { …` code in the source.

**Reply by mcmcclur** [reply]
Yes, I’m planning on posting the code to GitHub and expanding the description. I wanted to get it out, though, and have a ton of other things.

glad you like it!

**Reply by Fil** [reply]
It would be neat to also support [speaker notes](https://revealjs.com/speaker-view/).

**Reply by mcmcclur** [reply]
I've updated my framework project to help explain how to create D3 images that are responsive to slide changes. You can go directly to the last column of slides that explains the transitions here:

https://mcmcclur.observablehq.cloud/mark-on-framework/posts/RevealJS/#/5

I've also pushed the code to GitHub:

https://github.com/mcmcclur/MarkOnFramework/tree/main/src/posts/RevealJS

**Reply by Fil** [reply]
Thanks! I've created a minimalist example at https://observablehq.observablehq.cloud/pangea/party/revealjs (it links to your site for people who want more details). Differences are I'm using the `npm:` specifiers instead of `npm add`, and I added support for dark mode.
